### DISCUSSION TITLE
gotox现在使用的是什么tls吗？如果要配置sni，如何配置。

### DISCUSSION THREAD

**Discussion by wpyok500** [SeaHOH/GotoX#59] [问答/咨询]
gotox现在使用的是什么tls吗？如果要配置sni，如何配置。

**Comment by SeaHOH** [comment]
没有现成的 QUIC 库，所以还是 TLS。IPv6 应该可以用。

使用 sni 上个 issues 就有讲 https://github.com/SeaHOH/GotoX/issues/58#issuecomment-348825796 。

**Reply by SeaHOH** [reply]
现在有现成的库了，但是我没时间。。。

**Comment by wpyok500** [comment]
不能像默认那样，直接在data文件夹底下配置一个sni_all.txt的sni   ip列表吗？

**Comment by wpyok500** [comment]
1、我将ip.txt内的ip全部替换成sni  ip 也是可以使用的。
2、我将ip.txt内的ip全部替换成sni  ip +配置一个servername  也是可以使用的。

**Comment by SeaHOH** [comment]
可以直接放到 ip.txt 使用。

**Comment by wpyok500** [comment]
gotox在我这解析一个网站速度有点慢，请问如何提高。

**Comment by SeaHOH** [comment]
换 DNS 或更改 DNS 服务器设置？

**Comment by blob2015** [comment]
大海兄，搭车问一下，估摸着啥时候会有人搞出来个quic库？
quic好像地位要提高啊。

**Comment by SeaHOH** [comment]
我想了个歪招：用 Go 写一个服务器来转换。

**Comment by blob2015** [comment]
大海兄强大

**Comment by Peytonlan** [comment]
如果是为了翻墙, 私以为quic上不宜投入太大精力, 我这quic封锁蛮长时间了

**Comment by SeaHOH** [comment]
只是有想法，根本没动力~

**Comment by blob2015** [comment]
>QUIC……已交给 IETF，预计将于 2018 年标准化
>TLS 1.3 已经到达标准化的最后阶段

tls 1.3＋quic，会产生一些变化，虽然到时候墙也会有变化，目前可以观望一阵子

@Peytonlan 
gp开启quic不能用？俺这里用得好好的啊，重灾区

**Comment by Peytonlan** [comment]
@blob2015
我这udp一直是间歇性中断, 一开始udp没断时, goproxy使用quic能够连接
再后来, 即使udp没断, goproxy也无法使用quic连接, 算是彻底封了

**Comment by blob2015** [comment]
@Peytonlan 
这个俺这边是这样的，就是有段时间，ip会变得很少很少，连接还是能连接的，quic
使用quic，ip数量很重要，最好以万计……同时不要添加一些个无效ip……
当然，也可能各地各isp状况不一样，真有可能有那家把quic给彻底封了也说不好

**Comment by Peytonlan** [comment]
goproxy有些天没用了, 怕我说的太主观, 相对认真重新试了下
暂时收回前面说的话, 不是因为情况有很大改善, 而是觉得goproxy的quic应该还有提升空间, 有不确定性

ip以万计, 谈何容易
gscan-quic 3.6, 默认数据库, 找出2个ip
gscan-quic 2.0, 默认数据库, 找出8个ip
2.0似乎搜索结果略好些
gscan-quic 2.0, 以前用goagent时积累的218万的ip库, 找出147个ip
算是凑出了157个ip用来测试
投入goproxy， 不超一分钟就连不上了
也容易理解, ip本身就少, 而且goproxy还使用了 clients3.google.com 这个在我这儿找死的域名来测连通性
以前懒就没折腾, 这次勤快点, 改两行代码, 换个域名
https://github.com/goproxy0/goproxy/blob/39611867adb372fa95ff8643d4e07cf357320d0e/httpproxy/filters/gae/gae.go#L372
https://github.com/goproxy0/goproxy/blob/39611867adb372fa95ff8643d4e07cf357320d0e/httpproxy/filters/gae/gae.go#L394
改了域名后, 的确有改善, 差不多放个两三首MV没问题, 然后继续断连, 还是报udp超时
虽然断了, 从输出看, 有个不正常的现象, good_addrs并没有下降多少, 有时甚至还有上升, 怀疑ip判定及调度有些问题
那就同时运行两个goproxy吧, 果然其中一个断了后, 另一个并没有同时中断, 这说明了不完全是ip不能连接导致的中断
暂时没时间也没精力折腾了, 到此为止

这次勤快点, 还是有个小收获的, 以前因为 clients3.google.com 的问题, goproxy配置sniproxy也运行不长时间的, 总算是稳定了

**Comment by blob2015** [comment]
有个讨论的issue，社长也参与，这会儿找不到了，好像还是在主任那边，记得他有说，蜗牛角上争天地，石火光中寄此身（白居易原诗修改而来？）。感觉你可能还是过于精细了，所谓的ip质量（带宽、响应速度）已经没有意义了。
粗茶淡饭保平安。
数量就是质量。

**Comment by Peytonlan** [comment]
问题是没数量啊, 又哪来的粗茶淡饭
火中取栗, 不精细点怎么办, 所以我对quic持保留态度

**Comment by blob2015** [comment]
你都精细了，那还是什么粗茶淡饭？这个暂时告一段落吧。
quic被N多人看淡，但都成了基础协议了，怎么看淡？我用quic，都没断过一天……

**Comment by blob2015** [comment]
@SeaHOH 
大海兄，最近你狂下视频？ :grinning:

**Comment by SeaHOH** [comment]
没，日常杀时间，顺手修点小 bug（复杂的不会）。

**Comment by blob2015** [comment]
@SeaHOH @Just2co 那天俺也试着学点，这年头，就不知道那年那月会完成这档子事了。好像不学点这个，都整不利索很多事……

**Comment by blob2015** [comment]
@Just2co 老兄说的，甚好，有实操经验的人哪。之前学正则，就是靠着修改小说，慢慢练出来的。感谢。

**Comment by SeaHOH** [comment]
时间很重要啊，之前我是太闲了。 :trollface:

**Comment by SeaHOH** [comment]
servername 的处理，你看这里 ec064a1794506cf8d5662a2177507201f870d92c

**Comment by Just2co** [comment]
@SeaHOH 如果访问的是谷歌域名，servername就是fonts.googleapis.com，如果是其他域名，就是xxx.appspot.com，但即便知道这样，也还是不知道怎么把goproxy相关脚本改过来。在这里的判断以前是没有的吧？

@Peytonlan 你熟悉goproxy，你能不能把这个问题修好，传给我用用？

**Comment by SeaHOH** [comment]
https://github.com/SeaHOH/goproxy/blob/master/httpproxy/filters/gae/gae.go#L123
这个是默认值

https://github.com/SeaHOH/goproxy/blob/master/httpproxy/helpers/dialer2.go#L115
从这儿开始分别处理

**Comment by wpyok500** [comment]
@Just2co 你修改的gop有在github发布吗？

**Comment by Just2co** [comment]
@wpyok500 我要是有这么大的本事倒好了，@Peytonlan 他好像会go语，让他改改

**Comment by SeaHOH** [comment]
@MeABc
问问进度如何，servername 的处理能一起修复了吗？

**Comment by MeABc** [comment]
@SeaHOH 呃。。。我没有看明白你们上面说的。。
另，我尽快整理上传代码，啊！！！！！！！！！要改的有点多啊！！！

**Comment by SeaHOH** [comment]
我说的进度是你上次更新的，不是这个啊。 :sweat_smile:

**Comment by MeABc** [comment]
SNI proxy 不能用吗? 还是放在 goproxy 里不能用? 这个要区分。

`如果访问的是谷歌域名，servername就是fonts.googleapis.com，如果是其他域名，就是xxx.appspot.com，` 

首先，googleapis or appspot 这种域名可能会被墙。另一个还要考虑到 sni proxy 是否过滤 google 等等。因素太多，让代理来做判断是不是有点复杂了，这个还是 “人工筛选”ip 是最多的。。。。

**Comment by MeABc** [comment]
人工筛选”ip 是最好的。。

**Comment by SeaHOH** [comment]
https://www.solidot.org/story?sid=56211
没法筛选，谷歌直接不干了。

参考我上面的处理 https://github.com/SeaHOH/GotoX/issues/59#issuecomment-381982966

**Comment by SeaHOH** [comment]
突然发现那家伙把 comment 删掉了，看起来就像我在自言自语。。。

**Comment by wpyok500** [comment]
@MeABc 在这里先感谢接手开发，辛苦了。

**Comment by linux40** [comment]
quic是怎么用？放到ip.txt里面么？

**Comment by SeaHOH** [comment]
>quic是怎么用？

暂不支持

**Comment by linux40** [comment]
今天自己扫的sni都挂了，新的也扫不出来，这几天忙完了试试goproxy（换这个工具前一直用的goagent，还坚持到了19大。。。）

**Comment by SeaHOH** [comment]
172.217.0.0/16
216.58.0.0/16
108.177.0.0/16

试试

**Comment by wpyok500** [comment]
只扫到四个
216.58.227.71
216.58.227.70
216.58.227.68
216.58.227.69
